### PR TITLE
Advanced setting service and model

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/core/models/common/advanced-settings.model.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export interface ExpenseFilterResponse extends ExpenseFilter {
+    id: number,
+    created_at: Date,
+    update_at: Date,
+    workspace: number
+}
+
+export type ExpenseFilterGetResponse = {
```
[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/core/models/common/advanced-settings.model.ts:29]: i already have a type with this name

**Comment by DhaaraniCIT** [comment]
PR comments are fixed in https://github.com/fylein/fyle-integrations-settings-app/pull/263

File: src/app/core/models/common/advanced-settings.model.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    condition: string,
+    operator: string,
+    values: string | string[]
+    rank: number,
+    join_by: string | null,
+    is_custom: boolean,
+    custom_field_type: string | null
+}
+
+export interface ExpenseFilterResponse extends ExpenseFilter {
```
Comment (ashwin1111) [inline src/app/core/models/common/advanced-settings.model.ts:22]: make this ExpenseFilter

File: src/app/core/models/common/advanced-settings.model.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    name: string;
+}
+
+export type ConditionField = {
+  field_name: string;
+  type: string;
+  is_custom: boolean;
+};
+
+export type ExpenseFilter = {
```
Comment (ashwin1111) [inline src/app/core/models/common/advanced-settings.model.ts:12]: make this ExpenseFilterPost

File: src/app/core/models/common/advanced-settings.model.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/core/models/common/advanced-settings.model.ts:29]: Screenshot 2023-11-15 at 7 23 55 PM

[the remark this answers is not in the document]
Comment (DhaaraniCIT) [reply src/app/core/models/common/advanced-settings.model.ts:29]: we already have ExpenseFIlter

**Pull request merged**